src/components/molecules/Carousel/Component.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useRef, useState, forwardRef, FC } from 'react';
+import {
+  FlatList,
+  View,
+  ListRenderItemInfo,
+  ViewToken,
+  Image,
+  LayoutChangeEvent,
+  ViewStyle,
+  ImageResizeMode,
+  StyleSheet,
+  ImageSourcePropType,
+} from 'react-native';
+import type { PropsWithRef } from '../../../type';
+import CarouselIndicator, { IndicatorStyle } from './Indicator';
+
+export type CarouselProps = PropsWithRef<{
+  separatorWidth?: number;
+  data: Array<ImageSourcePropType>;
+  imageResizeMode?: ImageResizeMode;
+  decelerationRate?: 'fast' | 'normal' | number;
+  indicatorStyles?: IndicatorStyle;
+  style?: ViewStyle;
+}>;
+
+const Carousel: FC<CarouselProps> = forwardRef<any, CarouselProps>(
+  (
+    {
+      style,
+      separatorWidth = 0,
+      data,
+      imageResizeMode = 'stretch',
+      decelerationRate = 'fast',
+      indicatorStyles,
+    },
+    ref
+  ) => {
+    const [index, setIndex] = useState<number>(0);
+    const slider = useRef<FlatList>(null);
+    const [totalItemWidth, setTotalItemWidth] = useState<number>();
+
+    const onViewableItemsChangedRef = useRef(
+      (info: {
+        viewableItems: Array<ViewToken>;
+        changed: Array<ViewToken>;
+      }) => {
+        const { viewableItems } = info;
+
+        if (viewableItems.length > 0) {
+          setIndex(viewableItems[0].index || 0);
+        }
+      }
+    );
+
+    const viewabilityConfigRef = useRef({
+      viewAreaCoveragePercentThreshold: 50,
+    });
+
+    const renderSeparator = () => <View style={{ width: separatorWidth }} />;
+    const renderItem = (params: ListRenderItemInfo<ImageSourcePropType>) => {
+      return (
+        <View
+          style={StyleSheet.flatten([
+            { width: totalItemWidth },
+            styles.containerItem,
+          ])}
+        >
+          <Image
+            resizeMode={imageResizeMode}
+            style={styles.imageItem}
+            source={params.item}
+          />
+        </View>
+      );
+    };
+    const handleLayout = (e: LayoutChangeEvent) => {
+      const {
+        nativeEvent: {
+          layout: { width },
+        },
+      } = e;
+
+      setTotalItemWidth(width + separatorWidth);
+    };
+
+    return (
+      <View ref={ref} style={style} onLayout={handleLayout}>
+        <FlatList
```

Comment:
it ok to use the library here, cause it has animation and something useful.

Reply:
@zrg-team You mean we can use external library rather than implementing by ourself?

Reply:
@zrg-team: any suggestion library?

Reply:
https://github.com/meliorence/react-native-snap-carousel
i think this is the most popular lib

Reply:
@zrg-team: ya, it's look good, but the last commit is last year and a lot of issues still pending. Does author still maintain this package?

Reply:
the package is stable i think

Reply:
https://github.com/meliorence/react-native-snap-carousel/pull/678#issuecomment-937575203

Reply:
anyway, we can choose another lib, i think there are a lot library out there

Reply:
Okay if it stable then let go with this one

Reply:
Maintaining a library like this will take us a lot of time. cause requirement will have animation, etc